here-naksha-app-service/src/test/java/com/here/naksha/app/service/NakshaAppTest.java

```diff
@@ -204,36 +205,104 @@ void tc0006_testGetStorageByWrongId() throws Exception {
     final String streamId = UUID.randomUUID().toString();
 
     // 2. Perform REST API call
     final HttpRequest request = HttpRequest.newBuilder(stdHttpRequest, (k, v) -> true)
         .uri(new URI(NAKSHA_HTTP_URI + "hub/storages"))
-        .POST(HttpRequest.BodyPublishers.ofString(bodyJson))
+        .GET()
         .header(HDR_STREAM_ID, streamId)
         .build();
     final HttpResponse<String> response = httpClient.send(request, HttpResponse.BodyHandlers.ofString());
 
     // 3. Perform assertions
-    assertEquals(400, response.statusCode(), "ResCode mismatch");
+    assertEquals(200, response.statusCode(), "ResCode mismatch");
     JSONAssert.assertEquals(
-        "Expecting failure response", expectedBodyPart, response.body(), JSONCompareMode.LENIENT);
+        "Expecting previously created storage", expectedBodyPart, response.body(), JSONCompareMode.LENIENT);
     assertEquals(streamId, getHeader(response, HDR_STREAM_ID), "StreamId mismatch");
   }
 
   @Test
   @Order(2)
-  void tc0008_testCreateEventHandler() throws Exception {
+  void tc0060_testUpdateStorage() throws Exception {
+    // Test API : PUT /hub/storages/{storageId}
+    // Given:
+    final String updateStorageJson = loadFileOrFail("TC0060_updateStorage/update_storage.json");
+    final String expectedRespBody = loadFileOrFail("TC0060_updateStorage/response.json");
+    final String streamId = UUID.randomUUID().toString();
+
+    // When:
+    final HttpRequest request = HttpRequest.newBuilder(stdHttpRequest, (k, v) -> true)
+        .uri(new URI(NAKSHA_HTTP_URI + "hub/storages/um-mod-dev"))
+        .PUT(HttpRequest.BodyPublishers.ofString(updateStorageJson))
+        .header(HDR_STREAM_ID, streamId)
+        .build();
+    final HttpResponse<String> response = httpClient.send(request, HttpResponse.BodyHandlers.ofString());
+
+    // Then:
+    assertEquals(200, response.statusCode());
+    JSONAssert.assertEquals(expectedRespBody, response.body(), JSONCompareMode.LENIENT);
+    assertEquals(streamId, getHeader(response, HDR_STREAM_ID));
+  }
+
+  @Test
+  @Order(2)
+  void tc0061_testUpdateNonexistentStorage() throws Exception {
+    // Test API : PUT /hub/storages/{storageId}
+    // Given:
+    final String updateStorageJson = loadFileOrFail("TC0061_updateNonexistentStorage/update_storage.json");
+    final String expectedErrorResponse = loadFileOrFail("TC0061_updateNonexistentStorage/response.json");
+    final String streamId = UUID.randomUUID().toString();
+
+    // When:
+    final HttpRequest request = HttpRequest.newBuilder(stdHttpRequest, (k, v) -> true)
+        .uri(new URI(NAKSHA_HTTP_URI + "hub/storages/this-id-does-not-exist"))
+        .PUT(HttpRequest.BodyPublishers.ofString(updateStorageJson))
+        .header(HDR_STREAM_ID, streamId)
+        .build();
+    final HttpResponse<String> response = httpClient.send(request, HttpResponse.BodyHandlers.ofString());
+
+    // Then:
+    assertEquals(409, response.statusCode());
```

Comment:
If we access non-existing storage, I think, it should return 404 ... perhaps need to check if mockhub needs to return proper result for no-data-found scenario during update.

Reply:
I have asked @xeus2001 to share his thoughts on what should be response from lib-psql when we try to Update feature which doesn't exist?
Will it:
1. throw SQLException with `PSQLState.NO_DATA` ?
2. throw SQL Exception with `PSQLState.UNIQUE_VIOLATION` ?
3. it will return ErrorResult with `XyzError.NOT_FOUND` ?

We need clear way to differentiate exception.
My recommendation is (1), but depending on what Alex confirms, @Amaneusz  - accordingly we can change `NHAdminWriterMock.updateFeature()` to behave on "no data found" scenario.

Reply:
I agree that `404` would be more suitable here. I will take a look at it

Reply:
Actually, there are two general kind of errors, when it comes to feature modification in a collection. The one is, some basic thing is wrong (collection does not exist) or a connection is broken or anything other fatal, something so we can't even start the processing. In that case I would return an `ErrorResult` with `reason` set to a corresponding XyzError (we need to make more constants, but that should not be an issue). The message will hold a human readable description of the error (for example "collection 'foo' does not exist").
The second kind of error is one that happens while we process the operations. As you can add multiple operations to a single `WriteFeatures` request I personally think we should return for every operation always one row with the result. Just, the action `EExecutedOp` may reflect an error, for example instead of `CREATED` there could be `FAIL` in it and you can query for the failure reason. If can as well provide the first error in some static function, if you anyway want to abort for the first error. However, this would allow you to ignore the errornous operations and still commit, returning the correct information or select all failed operations and just return them or something like this.
What do you think?

Reply:
I think that in requests where we have multiple operations (`WriteFeatures`) we should return result for each of the operation so that client can decide whether retry is necessary for the failed ones or note.
For this particular case (single operation in `WriteFeatures` class) I would simply add another type - `WriteFeature` which will always hold only single feature operation - easier to interpret and also quite self explanatory :)

Reply:
I created separate ticket for this: https://devzone.it.here.com/jira/browse/MCPODS-6576
I know it's a small topic but these changes are needed to be merged so others can proceed with their changes. I will pick this task as soon as we decide on something, for now we're sticking with 409 we know we want to change.